### PR TITLE
Change UseSpecificHttpStatusCodes to should violation

### LINKED ISSUE #374: Response status code check is too strict
#313/#340 added a check that verified that each operation only returns one of the status codes listed in http://zalando.github.io/restful-api-guidelines/http/Http.html#must-use-specific-http-status-codes (it also checks that the HTTP method fits).

I don't think that rule is meant to be understood as "don't ever use any other status code than those listed here" (the list says "Our list of most commonly used and best understood HTTP status codes") – so a violation against this should not be a "MUST" violation.

Also, it disallows the use of "default", which (with a response-body of a problem object) is IMHO fine.

### PR DESCRIPTION
**Pull request by fmueller**
Closes #374 

It's a quick fix until Zalando's API guidelines were adapted accordingly.

### CODE REVIEW

File: server/src/main/java/de/zalando/zally/rule/UseSpecificHttpStatusCodes.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ class UseSpecificHttpStatusCodes(@Autowired rulesConfig: Config) : AbstractRule(
     override val title = "Use Specific HTTP Status Codes"
     override val url = "http://zalando.github.io/restful-api-guidelines/http/Http.html" +
         "#must-use-specific-http-status-codes"
-    override val violationType = ViolationType.MUST
+    override val violationType = ViolationType.SHOULD
     override val code = "M016"
```
Comment (netme) [inline server/src/main/java/de/zalando/zally/rule/UseSpecificHttpStatusCodes.kt]: Should the code also be also changed to `S`?

Comment (netme) [inline server/src/main/java/de/zalando/zally/rule/UseSpecificHttpStatusCodes.kt]: Or maybe some comment should be added, now it's a bit confusing :-)

Comment (fmueller) [inline server/src/main/java/de/zalando/zally/rule/UseSpecificHttpStatusCodes.kt]: I'll add a comment in the code and reference the issue.

**Comment by fmueller** [comment]
@netme Added comment, can you please re-check it? :)

**Comment by netme** [comment]
👍 

Thanks for adding a comment

**Comment by netme** [comment]
Restarted builds, let's see if it will help

**Comment by fmueller** [comment]
This won't help because I have to adapt our integration tests. ;-)

**Comment by netme** [comment]
👍

**Comment by netme** [comment]
Thanks for fixing

**Pull request merged**